rskj-core/src/main/java/co/rsk/crypto/Keccak256.java

```diff
@@ -18,36 +18,54 @@
 
 package co.rsk.crypto;
 
-import com.google.common.primitives.Ints;
-import co.rsk.bitcoinj.core.Utils;
+import org.ethereum.crypto.HashUtil;
+import org.ethereum.util.RLP;
+import org.spongycastle.util.encoders.Hex;
 
 import java.io.Serializable;
 import java.math.BigInteger;
 import java.util.Arrays;
 
-import static com.google.common.base.Preconditions.checkArgument;
+import static org.ethereum.util.ByteUtil.EMPTY_BYTE_ARRAY;
 
 /**
- * A Sha3Hash just wraps a byte[] so that equals and hashcode work correctly, allowing it to be used as keys in a
+ * A Keccak256 just wraps a byte[] so that equals and hashcode work correctly, allowing it to be used as keys in a
  * map. It also checks that the length is correct and provides a bit more type safety.
  */
-public class Sha3Hash implements Serializable, Comparable<Sha3Hash> {
+public class Keccak256 implements Serializable, Comparable<Keccak256> {
```

Comment:
There are a few things I would review about this class.

1. Does it need to be Serializable?

2. It is called Keccak256, but there's absolutely nothing that forces you to use a Keccak256 hash or to even create one from a any byte[]. Constructors also do not check that the byte[] is 32 bytes long.

3. As keccak should be immutable, it sounds like constructor should copy the byte[] and getbytes() return a copy. Then use equals() or compareTo() when neeeded, instead of comparing the getBytes() result

Reply:
Even better would be to create classes that inherit from this (without changing anything at all, probably) that should be "BlockHash", "TxHash", "StateRootHash", etc.

Reply:
I think that without something like this last thing, there's no much benefit of having a Keccak256 class over the ByteArrayWrapper we were using.